pkg/controller/render/render_controller.go

```diff
@@ -636,3 +724,521 @@ func getMachineConfigsForPool(pool *mcfgv1.MachineConfigPool, configs []*mcfgv1.
+
+	# Apply the butane-rendered ignition so it ends up in origin.d 
+	# machineconfig -> ignition -> ignition-liveapply -> butane file -> more ignition -> ignition-liveapply -> treefile -> rebuild
+	RUN ignition-liveapply /etc/mco-butane.json 
+
+	# Rebuild origin.d (I included an /etc/yum.repos.d/ file in my machineconfig so it could find the RPMS, that's why this works)
+	RUN rpm-ostree ex rebuild && rm -rf /var/cache /etc/rpm-ostree/origin.d
+
+
+	# This is so we can get the machineconfig injected
+	ARG machineconfig=unknown
+	# Apply the injected machineconfig name as a label so node_controller can check it
+	LABEL machineconfig=$machineconfig
+	`
+
+	_, err := ctrl.buildclient.BuildV1().BuildConfigs(ctrlcommon.MCONamespace).Get(context.TODO(), buildConfigName, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+
+		// Construct a buildconfig for this pool if it doesn't exist
+		buildConfig := &buildv1.BuildConfig{
+
+			TypeMeta: metav1.TypeMeta{},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      buildConfigName,
+				Namespace: ctrlcommon.MCONamespace,
+				Annotations: map[string]string{
+					"machineconfiguration.openshift.io/pool": pool.Name,
+				},
+			},
+			Spec: buildv1.BuildConfigSpec{
+				RunPolicy: "Serial",
+				// Simple dockerfile build, just the text from above
+				CommonSpec: buildv1.CommonSpec{
+					Source: buildv1.BuildSource{
+						Type:       "Dockerfile",
+						Dockerfile: &dockerFile,
+					},
+					Strategy: buildv1.BuildStrategy{
+						DockerStrategy: &buildv1.DockerBuildStrategy{},
+						Type:           "Docker",
+					},
+					// Output to the imagestreams we made before
+					Output: buildv1.BuildOutput{
+						To: &corev1.ObjectReference{
+							Kind: "ImageStreamTag",
+							Name: contentImageStreamName + ":latest",
+						},
+						// TODO(jkyros): I want to label these images with which rendered config they were built from
+						// but there doesn't seem to be a way to get it in there easily
+					},
+				},
+			},
+		}
+
+		// Set the owner references so these get cleaned up if the pool gets deleted
+		poolKind := mcfgv1.SchemeGroupVersion.WithKind("MachineConfigPool")
+		oref := metav1.NewControllerRef(pool, poolKind)
+		buildConfig.SetOwnerReferences([]metav1.OwnerReference{*oref})
+
+		// Create the buildconfig
+		_, err := ctrl.buildclient.BuildV1().BuildConfigs(ctrlcommon.MCONamespace).Create(context.TODO(), buildConfig, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+
+		glog.Infof("Buildconfig %s has been created for pool %s", buildConfigName, pool.Name)
+	} else if err != nil {
+		// some other error happened
+		return err
+	} else {
+		// already existed, do nothing
+		glog.Infof("Buildconfig %s already existed for pool %s", buildConfigName, pool.Name)
+	}
+
+	return nil
+}
+
+// experimentalRenderToImagestream is a dirty hack that sets up imagestream rendering of config (hopefully) without interrupting normal operation :)
```

Comment:
When I hear render I think render config as in what `generateRenderedMachineConfig` does. Not sure if that's actually true, but I think it might be more clear if we used a different word for building an image. I also think that might help clarify how this function is building two images